Kudu.Core/Settings/DiagnosticsSettingsManager.cs

```diff
@@ -87,6 +93,7 @@ private void SaveSettings(DiagnosticsSettings diagnosticsSettings)
                 FileSystemHelpers.EnsureDirectory(Path.GetDirectoryName(_path));
             }
 
+            diagnosticsSettings.EnsureValid();
             string fileContent = JsonConvert.SerializeObject(diagnosticsSettings);
```

Comment:
Looking at the code, diagnosticsSettings.EnsureValid() is not necessary as JsonConvert.SerializeObject(diagnosticsSettings); and DiagnosticsSettings class member type would have enforce it already.  No?  Try simulate it - let me know if you can get pass this without exception.